code/_onclick/hud/action_button.dm

```diff
@@ -14,8 +15,9 @@
 		moved = 0
 		usr.update_action_buttons() //redraw buttons that are no longer considered "moved"
 		return 1
-	if(usr.next_move >= world.time) // Is this needed ?
```

Comment:
I'd prefer to reuse the same shit so we can just test for and reset the same var everywhere rather than needing to set 4 vars which effectively do the same thing.